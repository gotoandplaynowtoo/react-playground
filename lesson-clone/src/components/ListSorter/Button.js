import Button from 'components/Button';

const XButton = Button.extend`
  border: 1px solid ${ props => props.focused ? '#222' : '#aaa'};
`;

export default XButton;
