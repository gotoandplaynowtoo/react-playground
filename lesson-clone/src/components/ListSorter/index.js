import React from 'react';
import Button from 'components/ListSorter/Button';

const ListSorter = (props) => {

  const { 
    activeSort,
    onClick 
  } = props;

  return (
    <div>
      Sort by: 
      <Button value='title' focused={activeSort === 'title'} onClick={onClick}>Title</Button>
      <Button value='author' focused={activeSort === 'author'} onClick={onClick}>Author</Button>
      <Button value='points'  focused={activeSort === 'points'} onClick={onClick}>Points</Button>
    </div>
  );
};

export default ListSorter;