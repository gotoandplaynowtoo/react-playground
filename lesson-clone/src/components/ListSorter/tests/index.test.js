import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ListSorter from 'components/ListSorter';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ListSorter/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer.create(<ListSorter/>).toJSON();
  expect(tree).toMatchSnapshot();
});