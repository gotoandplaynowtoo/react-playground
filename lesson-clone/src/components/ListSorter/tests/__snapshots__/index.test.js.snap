// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  Sort by:
  <button
    className="sc-bwzfXH CKSnV"
    onClick={undefined}
    value="title"
  >
    Title
  </button>
  <button
    className="sc-bwzfXH CKSnV"
    onClick={undefined}
    value="author"
  >
    Author
  </button>
  <button
    className="sc-bwzfXH CKSnV"
    onClick={undefined}
    value="points"
  >
    Points
  </button>
</div>
`;
